sources/TileDB.CSharp/Enums.cs

```diff
@@ -1,179 +1,756 @@
+        TILEDB_TIME_SEC = TimeSecond,
+        [Obsolete("Use TimeMillisecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_MS = TimeMillisecond,
+        [Obsolete("Use TimeMicrosecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_US = TimeMicrosecond,
+        [Obsolete("Use TimeNanosecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_NS = TimeNanosecond,
+        [Obsolete("Use TimePicosecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_PS = TimePicosecond,
+        [Obsolete("Use TimeFemtosecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_FS = TimeFemtosecond,
+        [Obsolete("Use TimeAttosecond instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_TIME_AS = TimeAttosecond,
+        [Obsolete("Use Blob instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_BLOB = Blob,
+        [Obsolete("Use Boolean instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_BOOL = Boolean
     }
 
+    /// <summary>
+    /// Specifies the type of an <see cref="Array"/>.
+    /// </summary>
     public enum ArrayType : uint
     {
-        TILEDB_DENSE = tiledb_array_type_t.TILEDB_DENSE,
-        TILEDB_SPARSE = tiledb_array_type_t.TILEDB_SPARSE,
+        /// <summary>
+        /// The array is dense.
+        /// </summary>
+        Dense = tiledb_array_type_t.TILEDB_DENSE,
+        /// <summary>
+        /// The array is sparse.
+        /// </summary>
+        Sparse = tiledb_array_type_t.TILEDB_SPARSE,
+        [Obsolete("Use Dense instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_DENSE = Dense,
+        [Obsolete("Use Sparse instead."), EditorBrowsable(EditorBrowsableState.Never)]
+        TILEDB_SPARSE = Sparse
     }
 
+    /// <summary>
+    /// Specifies the order of data stored in an <see cref="Array"/>
+    /// or sent or retrieved from a <see cref="Query"/>.
+    /// </summary>
     public enum LayoutType : uint
     {
-        TILEDB_ROW_MAJOR = tiledb_layout_t.TILEDB_ROW_MAJOR,
-        TILEDB_COL_MAJOR = tiledb_layout_t.TILEDB_COL_MAJOR,
-        TILEDB_GLOBAL_ORDER = tiledb_layout_t.TILEDB_GLOBAL_ORDER,
-        TILEDB_UNORDERED = tiledb_layout_t.TILEDB_UNORDERED,
-        TILEDB_HILBERT = tiledb_layout_t.TILEDB_HILBERT,
+        /// <summary>
+        /// Data are in row-major order.
+        /// </summary>
+        RowMajor = tiledb_layout_t.TILEDB_ROW_MAJOR,
+        /// <summary>
+        /// Data are in column-major order.
+        /// </summary>
+        ColumnMajor = tiledb_layout_t.TILEDB_COL_MAJOR,
+        /// <summary>
+        /// Data are in global order.
+        /// </summary>
+        GlobalOrder = tiledb_layout_t.TILEDB_GLOBAL_ORDER,
+        /// <summary>
+        /// Data are unordered.
+        /// </summary>
+        // TODO: Make them more explanatory.
```

Comment:
Just wanted to point out the `TODO` here in case you left it unintentionally. I use them too and accidentally commit them sometimes.

Reply:
I left it intentionally; I still can't figure them out after being explained what they are. I opened [SC-23104](https://app.shortcut.com/tiledb-inc/story/23104/better-explain-in-the-documentation-what-unordered-and-global-order-layouts-are) to update the documentation.